runtime/src/serde_snapshot/tests.rs

```diff
@@ -250,6 +267,12 @@ fn test_bank_serialize_style(serde_style: SerdeStyle) {
     assert_eq!(dbank.get_balance(&key1.pubkey()), 0);
     assert_eq!(dbank.get_balance(&key2.pubkey()), 10);
     assert_eq!(dbank.get_balance(&key3.pubkey()), 0);
+    // In case we are using older version, we expect it does not understand
+    // `prior_roots` field.  Thus, we reset the prior_roots to Vec::default
+    // and expect the bank2 == dbank.
+    if serde_style == SerdeStyle::Older {
+        bank2.prior_roots = Vec::<Slot>::default();
+    }
```

Comment:
This code will go inside older, specifically in the `from()` implementations. This is how older snapshot version handle (i.e. ignore) new fields.

Refer to how `UnusedAccounts` is handled in the old version. (This case is actually the opposite, since `unused_accounts` was removed, but hopefully the concept is illustrative enough to make sense 😊.) You'll so something similar for `prior_roots`, such that when deserializing an older snapshot, it sets the `BankFieldsToDeserialize::prior_roots` to `Vec::default()`.

Reply:
`bank2` is the original instance where I artificially change its prior_roots at line 221.  Pretending it is from new.

And `dbank` is the deserialized one, which its `prior_roots` is handled by its `from()`.

Since we expect `bank2` and `dbank` disagree with its `prior_roots`, I am resetting `bank2.prior_roots` to the expected value of `dbank`.

Reply:
Gotcha, thanks! I see now. I'll continue this thread up in https://github.com/solana-labs/solana/pull/23331#discussion_r814291191